sql-plugin/src/main/scala/org/apache/spark/sql/rapids/RapidsShuffleInternalManagerBase.scala

```diff
@@ -749,16 +747,14 @@ abstract class RapidsShuffleThreadedReaderBase[K, C](
     private def popFetchedIfAvailable(): Unit = {
       // If fetcherIterator is not exhausted, we try and get as many
       // ready results.
-      if (!pendingIts.isEmpty) {
+      if (pendingIts.nonEmpty) {
         var continue = true
-        var i = 0
-        while(pendingIts.size() > 0 && continue) {
-          val blockState = pendingIts.peek()
-          i = i + 1
+        while(pendingIts.nonEmpty && continue) {
+          val blockState = pendingIts(0)
           // check if we can handle the head batch now
           if (limiter.acquire(blockState.getNextBatchSize)) {
             // kick off deserialization task
-            pendingIts.remove(blockState)
+            pendingIts.remove(0)
```

Comment:
Since we're treating `pendingIts` like a queue, does ArrayBuffer make sense here?  Seems a queue would be more appropriate here, and if we want to keep it an array then ArrayDeque might be a good choice.